contracts/schemes/JoinAndQuit.sol

```diff
@@ -0,0 +1,243 @@
+
+
+/**
+ * @title A scheme for join in a dao.
+ * - A member can be proposed to join in by sending a min amount of fee.
+ * - A member can ask to quite (RageQuit) a dao on any time.
+ * - A member can donate to a dao.
+ */
+contract JoinAndQuit is
+        VotingMachineCallbacks,
+        ProposalExecuteInterface,
+        Initializable {
+    using SafeMath for uint;
+    using SafeERC20 for address;
+    using StringUtil for string;
+
+    event JoinInProposal(
+        address indexed _avatar,
+        bytes32 indexed _proposalId,
+        string _descriptionHash,
+        address _proposedMember,
+        uint256 _fundAmount
+    );
+
+    event FundedDeadLineReached(
+        address indexed _avatar
+    );
+
+    event Donation(
+        address indexed _avatar,
+        uint256 indexed _donation
+    );
+
+    event RageQuit(
+        address indexed _avatar,
+        uint256 indexed _refund
+    );
+
+    event RedeemReputation(
+        address indexed _avatar,
+        bytes32 indexed _proposalId,
+        address indexed _beneficiary,
+        uint256 _amount);
+
+    event ProposalExecuted(address indexed _avatar, bytes32 indexed _proposalId, int256 _decision);
+
+    struct Proposal {
+        bool accepted;
+        address proposedMember;
+        address funder;
+        uint256 funding;
+    }
+
+    mapping(bytes32=>Proposal) public proposals;
+    mapping(address=>uint256) public fundings;
+
+    IntVoteInterface public votingMachine;
+    bytes32 public voteParams;
+    Avatar public avatar;
+    IERC20 public fundingToken;
+    uint256 public minFeeToJoin;
+    uint256 public memberReputation;
+    uint256 public fundingGoal;
+    uint256 public fundingGoalDeadLine;
+    uint256 public totalDonation;
+    string public constant FUNDED_BEFORE_DEADLINE_KEY = "FUNDED_BEFORE_DEADLINE";
+    string public constant FUNDED_BEFORE_DEADLINE_VALUE = "TRUE";
+
+    /**
+     * @dev initialize
+     * @param _avatar the avatar this scheme referring to.
+     * @param _votingMachine the voting machines address to
+     * @param _voteParams voting machine parameters.
+     * @param _fundingToken the funding token
+     * @param _minFeeToJoin minimum fee required to join
+     * @param _memberReputation the repution which will be allocated for members
+     * @param _fundingGoal the funding goal
+     * @param _fundingGoalDeadLine the funding goal deadline
+     */
+    function initialize(
+        Avatar _avatar,
+        IntVoteInterface _votingMachine,
+        bytes32 _voteParams,
+        IERC20 _fundingToken,
+        uint256 _minFeeToJoin,
+        uint256 _memberReputation,
+        uint256 _fundingGoal,
+        uint256 _fundingGoalDeadLine
+    )
+    external
+    initializer
+    {
+        require(_avatar != Avatar(0), "avatar cannot be zero");
+        avatar = _avatar;
+        votingMachine = _votingMachine;
+        voteParams = _voteParams;
+        fundingToken = _fundingToken;
+        minFeeToJoin = _minFeeToJoin;
```

Comment:
If a DAO wanted to change this parameter, how would that work? Do we want to add a function to change these values? Or would that mean to remove this scheme and deploy a new one with different params?

Reply:
Yes.
using [SchemeFactory](https://github.com/daostack/arc/blob/arc-factory/contracts/schemes/SchemeFactory.sol) scheme .